sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.models.ts

```diff
@@ -0,0 +1,248 @@
+import {
+  IsDate,
+  IsEnum,
+  IsIn,
+  IsNumberString,
+  Length,
+  ValidateIf,
+  ValidationArguments,
+} from "class-validator";
+import {
+  APPLICATION_NUMBER_LENGTH,
+  ApplicationStatus,
+  SIN_NUMBER_LENGTH,
+} from "@sims/sims-db";
+import {
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalImportTextModel,
+  DataTextHeaders,
+} from "./application-bulk-withdrawal-import-text.models";
+import { IsValidSIN } from "../../utilities/class-validation";
+
+/**
+ * Possible warnings unique identifiers.
+ */
+export enum ApplicationWithdrawalValidationWarnings {
+  ApplicationNotFound = "applicationNotFound",
+  NoMatchingRecordFound = "noMatchingRecordFound",
+  HasPreviouslyBeenWithdrawn = "hasPreviouslyBeenWithdrawn",
+}
+
+/**
+ * Types of non-critical errors that will not prevent
+ * the offering from being saved.
+ */
+export enum ValidationContextTypes {
+  Warning = "warning",
+}
+
+/**
+ * Represent the context of an error that has an additional context to
+ * express a possible warning or some relevant information captured during
+ * the validation process.
+ * All validations when failed will generate an error. The ones that have
+ * the warning or information contexts will not be considered critical.
+ */
+export class ValidationContext {
+  /**
+   * Creates an error context that will make the error downgrade to
+   * a condition of a warning information will force the offering
+   * to be reviewed by the Ministry.
+   * @param warningTypeCode warning code that uniquely identifies this condition.
+   * @returns the warning context.
+   */
+  static CreateWarning(
+    warningTypeCode: ApplicationWithdrawalValidationWarnings,
+  ): ValidationContext {
+    const newContext = new ValidationContext();
+    newContext.type = ValidationContextTypes.Warning;
+    newContext.typeCode = warningTypeCode;
+    return newContext;
+  }
+
+  /**
+   * Context type.
+   */
+  type: ValidationContextTypes;
+  /**
+   * Unique identifier of the validation error.
+   */
+  typeCode: ApplicationWithdrawalValidationWarnings;
+}
+
+export class ApplicationBulkWithdrawalImportBusinessValidationModel {
+  /**
+   * SIN.
+   */
+  @IsValidSIN({
+    message: `${DataTextHeaders.sin} must be a valid SIN.`,
+  })
+  @Length(SIN_NUMBER_LENGTH, SIN_NUMBER_LENGTH, {
+    message: `${DataTextHeaders.sin} must be of length ${SIN_NUMBER_LENGTH}.`,
+  })
```

Comment:
Just wondering, this validation will come under line 77 right? so, do we need it?